scripts/format-icon-data.js

```diff
@@ -2,9 +2,88 @@
  * @file
  * Format _data/simple-icons.json.
  */
-import {getIconsData} from '../sdk.mjs';
+import {collator, getIconsData} from '../sdk.mjs';
 import {sortIconsCompare, writeIconsData} from './utils.js';
 
-const iconsData = await getIconsData();
-iconsData.sort(sortIconsCompare);
-writeIconsData(iconsData);
+/**
+ * @typedef {import("../sdk.d.ts").IconData} IconData
+ */
+
+/**
+ * Sort icon data obeject.
+ * @param {IconData} icon The icon data as it appears in *_data/simple-icons.json*.
+ * @returns {IconData} The sorted icon data.
+ */
+const sortIcon = (icon) => {
+	const keyOrder = [
+		'title',
+		'slug',
+		'hex',
+		'source',
+		'guidelines',
+		'license',
+		'aliases',
+		// This is not appears in icon data but it's in the alias object.
+		'loc',
+	];
+	const sortedIcon = Object.fromEntries(
+		Object.entries(icon).sort(
+			([key1], [key2]) => keyOrder.indexOf(key1) - keyOrder.indexOf(key2),
+		),
+	);
+	return sortedIcon;
+};
+
+/**
+ * Sort license object.
+ * @param {IconData['license']} license The license object as it appears in *_data/simple-icons.json*.
+ * @returns {IconData['license'] | undefined} The sorted license object.
+ */
+const sortLicense = (license) => {
+	if (!license) return undefined;
+	const keyOrder = ['type', 'url'];
+	const sortedLicense = Object.fromEntries(
+		Object.entries(license).sort(
+			([key1], [key2]) => keyOrder.indexOf(key1) - keyOrder.indexOf(key2),
+		),
+	);
+	return sortedLicense;
+};
+
+/**
+ * Sort object key alphabetically.
+ * @param {IconData['aliases']} aliases The aliases object as it appears in *_data/simple-icons.json*.
+ * @returns {IconData['aliases'] | undefined} The sorted aliases object.
+ */
+const sortAlphabetically = (aliases) => {
+	if (!aliases) return undefined;
+	const sortAliases = Object.fromEntries(
+		Object.entries(aliases).sort(([key1], [key2]) => (key1 > key2 ? 1 : -1)),
+	);
+	return sortAliases;
+};
+
+const icons = await getIconsData();
+const mappedIcons = icons.map((icon) => {
+	return sortIcon({
```

Comment:
Would you mind to share the same sorting implementation in the `fieldsSorted` function of *ourlint.js*?